python/cudf/cudf/core/column/string.py

```diff
@@ -2062,12 +2101,19 @@ def split(self, pat=None, n=-1, expand=None, **kwargs):
         if pat is None:
             pat = ""
 
-        result_table = cpp_split(self._column, as_device_scalar(pat, "str"), n)
-        if len(result_table._data) == 1:
-            if result_table._data[0].null_count == len(self._parent):
-                result_table = []
-            if self._column.null_count == len(self._column):
-                result_table = [self._column.copy()]
+        if expand:
+            result_table = cpp_split(
+                self._column, as_device_scalar(pat, "str"), n
+            )
+            if len(result_table._data) == 1:
+                if result_table._data[0].null_count == len(self._parent):
+                    result_table = []
+                if self._column.null_count == len(self._column):
+                    result_table = [self._column.copy()]
```

Comment:
Is it possible to do this before the libcudf call and return early?

Reply:
Done 👍